com.unity.sg2/Editor/GraphDeltaRegistry/Registry/ParametricTypeUtils.cs

```diff
@@ -0,0 +1,95 @@
+using UnityEngine;
+
+namespace UnityEditor.ShaderGraph.Defs
+{
+    internal static class ParametricTypeUtils
+    {
+        internal static string ParametricToHLSL(string name, int length, int height, float[] data)
+        {
+            string result = name + "(";
+
+            for (int i = 0; i < length * height; ++i)
+            {
+                result += $"{data[i]}";
+                if (i != length * height - 1)
+                    result += ", ";
+            }
+            result += ")";
+            return result;
+        }
+
+        private static void ManagedToParametric(object ovalue, out string name, out int length, out int height, out float[] data)
+        {
+            name = "float";
+            height = 1;
+            switch (ovalue)
+            {
+                default: throw new System.Exception("Type not supported.");
```

Comment:
This is a `private` method and the error message is not used. Can you message to the caller here with an output value, rather than pay the cost of throwing an Exception?

Reply:
Oh yea-- I didn't start it out as private, thanks.

Reply:
I have it returning a bool instead for now.